Sources/NativeBuilder/ContainerBuildCache/AsyncLock.swift

```diff
@@ -0,0 +1,56 @@
+//===----------------------------------------------------------------------===//
+// Copyright © 2025 Apple Inc. and the container project authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+/// `AsyncLock` provides cooperative async locking for serialized access to critical sections.
+/// This is particularly useful for atomic updates to cache.json and other shared resources.
+public actor AsyncLock {
```

Comment:
We can use the one from ContainerizationExtras (although doing a quick control-f here after opening quite a few of the diffs that you need to manually open, I don't see any files that reference AsyncLock yet)

Reply:
Thanks! It should be noted that there's a lot of placeholder code right now that can and will change shape as the implementation gets filled in. 

I'll remove this one since it's not even being referenced anywhere.